### PR TITLE
Add CVE scan GitHub workflow that is triggered on pull requests

### PR DESCRIPTION
**Pull request by weresch**
Thank you for contributing to the CF CLI! Please read the following:

* Please make sure you have implemented changes in line with the [contributing guidelines](https://github.com/cloudfoundry/cli/blob/main/.github/CONTRIBUTING.md)
* We're not allowed to accept any PRs without a signed CLA, no matter how small.
If your contribution falls under a company CLA but your membership is not public, expect delays while we confirm.
* All new code requires tests to protect against regressions.
* Contributions must be made against the appropriate branch. See the [contributing guidelines](https://github.com/cloudfoundry/cli/blob/main/.github/CONTRIBUTING.md)
* Contributions must conform to our [style guide](https://github.com/cloudfoundry/cli/wiki/CLI-Product-Specific-Style-Guide). Please reach out to us if you have questions.

## Description of the Change

Adding a CVE scan GitHub workflow on PRs.

## Why Is This PR Valuable?

The CVE scan GitHub workflow on PRs helps prevent known CVEs from being added to the codebase.

## Applicable Issues

No applicable issues

## How Urgent Is The Change?

Not urgent

## Other Relevant Parties

No one else

### CODE REVIEW

**Comment by weresch** [comment]
I paired on this work with @chinigo

File: .github/scripts/format-cve-scan-results.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+[[ "${TRACE:-0}" == "1" ]] && set -o xtrace
```
Comment (a-b) [inline .github/scripts/format-cve-scan-results.sh:3]: ❤️

**Comment by a-b** [comment]
Considering the volume of shell scripts we're writing, should we try https://github.com/shellspec/shellspec ?

File: .github/scripts/format-cve-scan-results.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    [${_table_underlines}],
+    (.matches[] | [
+        .artifact.name,
+        .artifact.version,
+        .vulnerability.fix.versions[0],
+        .artifact.type,
+        .vulnerability.id,
+        .vulnerability.severity
+    ]) | @tsv" "${_results_filename}" \
+    | sed 's/|/\\|/g' \
```
Comment (a-b) [inline .github/scripts/format-cve-scan-results.sh:45]: Have you consider something like `column --table`

File: .github/workflows/check-cves.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Print scan results
+      run: .github/scripts/format-cve-scan-results.sh results.json > $GITHUB_STEP_SUMMARY
+      if: always()
+
+    - name: Archive CVE scan results
+      uses: actions/upload-artifact@v4
+      if: always()
+      with:
+        name: cve-scan-results-${{ github.sha }}-${{ github.run_id }}-${{ github.run_number }}
+        path: results.json
```
Comment (a-b) [inline .github/workflows/check-cves.yml:32]: Adding a date or run number to the filename would make it easier for someone to download and work with multiple result artifacts in the future.

**Review by a-b — APPROVED** [review]
LGTM

**Pull request merged**